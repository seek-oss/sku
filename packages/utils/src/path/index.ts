export { toPosixPath } from './toPosixPath.js';
