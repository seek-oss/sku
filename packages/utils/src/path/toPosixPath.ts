import path from 'node:path';

/**
 * Replaces all win32 path separators (\) with posix path separators (/)
 */
export const toPosixPath = (inputPath: string) =>
  inputPath.replaceAll(path.win32.sep, path.posix.sep);
