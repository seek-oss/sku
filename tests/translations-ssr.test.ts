import assert from 'node:assert/strict';
import path from 'node:path';
import {
  runSkuScriptInDir,
  waitForUrls,
  getAppSnapshot,
} from '@sku-private/test-utils';

import skuSsrConfigImport from '@sku-fixtures/translations/sku-ssr.config.ts';
import type { ChildProcess } from 'node:child_process';
import { createRequire } from 'node:module';

const require = createRequire(import.meta.url);

const __dirname = path.dirname(new URL(import.meta.url).pathname);

const appDir = path.dirname(
  require.resolve('@sku-fixtures/translations/sku-ssr.config.ts'),
);

// TODO: fix this casting. Typescript is resolving the default export the whole `import` type.
const skuSsrConfig =
  skuSsrConfigImport as unknown as typeof skuSsrConfigImport.default;

assert(skuSsrConfig.serverPort, 'sku config has serverPort');
const getTestConfig = (skuConfig: typeof skuSsrConfig) => ({
  backendUrl: `http://localhost:${skuConfig.serverPort}`,
  targetDirectory: path.join(__dirname, skuConfig.target),
});

describe('ssr translations', () => {
  let server: ChildProcess;
  const { backendUrl } = getTestConfig(skuSsrConfig);

  beforeAll(async () => {
    server = await runSkuScriptInDir('start-ssr', appDir, [
      '--config=sku-ssr.config.ts',
    ]);
    await waitForUrls(backendUrl);
  });

  afterAll(async () => {
    await server.kill();
  });

  it('should render en', async () => {
    const app = await getAppSnapshot(`${backendUrl}/en`);
    expect(app).toMatchSnapshot();
  });

  it('should render fr', async () => {
    const app = await getAppSnapshot(`${backendUrl}/fr`);
    expect(app).toMatchSnapshot();
  });

  it('should render en-PSEUDO', async () => {
    const app = await getAppSnapshot(`${backendUrl}/en?pseudo=true`);
    expect(app).toMatchSnapshot();
  });
});
