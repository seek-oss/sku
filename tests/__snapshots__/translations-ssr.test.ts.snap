// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr translations should render en 1`] = `
SCRIPTS: [
  "http://127.0.0.1:8313/runtime.js",
  "http://127.0.0.1:8313/5.js",
  "http://127.0.0.1:8313/main.js",
  "http://127.0.0.1:8313/en-translations.js",
]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
      crossorigin="anonymous"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
      crossorigin="anonymous"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
      crossorigin="anonymous"
    >
    <link
      data-chunk="en-translations"
      rel="preload"
      as="script"
      href="scripts[3]"
      crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello sku
        Company name
      </div>
    </div>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
      crossorigin="anonymous"
    >
      [1]
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
      crossorigin="anonymous"
    >
      {"namedChunks":["en-translations"]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
      crossorigin="anonymous"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
      crossorigin="anonymous"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
      crossorigin="anonymous"
    >
    </script>
    <script
      async
      data-chunk="en-translations"
      src="scripts[3]"
      crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`ssr translations should render en-PSEUDO 1`] = `
SCRIPTS: [
  "http://127.0.0.1:8313/runtime.js",
  "http://127.0.0.1:8313/5.js",
  "http://127.0.0.1:8313/main.js",
  "http://127.0.0.1:8313/en-PSEUDO-translations.js",
]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
      crossorigin="anonymous"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
      crossorigin="anonymous"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
      crossorigin="anonymous"
    >
    <link
      data-chunk="en-PSEUDO-translations"
      rel="preload"
      as="script"
      href="scripts[3]"
      crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div>
        [Ḩẽẽẽƚƚööö šķǚǚǚ]
        [Çööm̂ƥăăกี้ýý กี้ăăm̂ẽẽ]
      </div>
    </div>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
      crossorigin="anonymous"
    >
      [0]
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
      crossorigin="anonymous"
    >
      {"namedChunks":["en-PSEUDO-translations"]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
      crossorigin="anonymous"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
      crossorigin="anonymous"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
      crossorigin="anonymous"
    >
    </script>
    <script
      async
      data-chunk="en-PSEUDO-translations"
      src="scripts[3]"
      crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`ssr translations should render fr 1`] = `
SCRIPTS: [
  "http://127.0.0.1:8313/runtime.js",
  "http://127.0.0.1:8313/5.js",
  "http://127.0.0.1:8313/main.js",
  "http://127.0.0.1:8313/fr-translations.js",
]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
      crossorigin="anonymous"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
      crossorigin="anonymous"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
      crossorigin="anonymous"
    >
    <link
      data-chunk="fr-translations"
      rel="preload"
      as="script"
      href="scripts[3]"
      crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Bonjour sku
        บริษัท
      </div>
    </div>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
      crossorigin="anonymous"
    >
      [2]
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
      crossorigin="anonymous"
    >
      {"namedChunks":["fr-translations"]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
      crossorigin="anonymous"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
      crossorigin="anonymous"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
      crossorigin="anonymous"
    >
    </script>
    <script
      async
      data-chunk="fr-translations"
      src="scripts[3]"
      crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;
